### PR TITLE
Give each SQL query in the shell its own context, and fix SIGTERM behavior for SQL shell

### LINKED ISSUE #871: Dolt handles NOW() incorrectly
Dolt seems to be reusing the session start time for successive values of NOW().

```
timestamps> create table a (x int primary key not null, t timestamp);
timestamps> insert into a values (2, now());
Query OK, 1 row affected
timestamps> insert into a values (3, now());
Query OK, 1 row affected
timestamps> insert into a values (4, now());
Query OK, 1 row affected
timestamps> insert into a values (4, now());
Query OK, 1 row affected
timestamps> select * from a;
+---+---------------------------------------+
| x | t                                     |
+---+---------------------------------------+
| 1 | 2020-09-11 21:53:24.7273339 +0000 UTC |
| 2 | 2020-09-11 21:57:14.3523038 +0000 UTC |
| 3 | 2020-09-11 21:57:14.3523038 +0000 UTC |
| 4 | 2020-09-11 21:57:14.3523038 +0000 UTC |
+---+---------------------------------------+
```

### PR DESCRIPTION
**Pull request by zachmu**
Fixes https://github.com/dolthub/dolt/issues/2101
Fixes https://github.com/dolthub/dolt/issues/2025
Fixes https://github.com/dolthub/dolt/issues/1155
Fixes #871

### CODE REVIEW

**Comment by zachmu** [comment]
Aaron, thoughts on this approach?

**Comment by zachmu** [comment]
Fixes https://github.com/dolthub/dolt/issues/871
Fixes https://github.com/dolthub/dolt/issues/2025

File: go/cmd/dolt/cli/command.go [hunk 1]
```diff
@@ -190,6 +204,14 @@ func (hc SubCommandHandler) handleCommand(ctx context.Context, commandStr string
 		ctx = events.NewContextForEvent(ctx, evt)
 	}
 
+	// Certain commands cannot tolerate a top-level signal handler (which cancels the root context) but need to manage
+	// their own interrupt semantics.
+	if signalCmd, ok := cmd.(SignalCommand); !ok || !signalCmd.InstallsSignalHandlers() {
+		var stop context.CancelFunc
+		ctx, stop = signal.NotifyContext(ctx, os.Interrupt)
```
Comment (reltuk) [inline go/cmd/dolt/cli/command.go:211]: `syscall.SIGTERM` as well?

**Review by reltuk — APPROVED** [review]
LGTM. One question.

File: go/cmd/dolt/cli/command.go (hunk 1 above)
Comment (zachmu) [reply go/cmd/dolt/cli/command.go:209]: Yeah probably good idea

**Pull request merged**